tests/testthat/helper-vdiffr.R

```diff
@@ -0,0 +1,86 @@
+# default is equal to whether NOT_CRAN is true or not
+enable_vdiffr <- identical(Sys.getenv("NOT_CRAN"), "true")
+
+# disable or enable vdiffr based on the state of USE_VDIFFR, if set
+if (identical(Sys.getenv("USE_VDIFFR"), "true")) {
+  enable_vdiffr <- TRUE
+} else if (identical(Sys.getenv("USE_VDIFFR"), "false")) {
+  enable_vdiffr <- FALSE
+}
+
+# disable vdiffr if version is old
+if (!requireNamespace("vdiffr", quietly = TRUE) ||
+    utils::packageVersion("vdiffr") < "0.2.3.9001") {
+  enable_vdiffr <- FALSE
+}
+
+# start up the image server and let vdiffr's svg writing method know about it
+if (enable_vdiffr) {
+  # generate random (port) number between 3000 & 8000
+  port <- floor(runif(1, 3001, 8000))
+  
+  # check to see if this port is usable
+  # TODO: this is how shiny does it -- is there a lighter-weight solution?
+  tmp <- try(httpuv::startServer("0.0.0.0", port, list()), silent = TRUE)
+  if (inherits(tmp, 'try-error')) {
+    stop(
+      "Tried to open orca server on port '", port, "', but it's not available. ", 
+      "Try running `test()` again"
+    )
+  } else {
+    httpuv::stopServer(tmp)
+  }
```

Comment:
It'd be great to have a lighter-weight way to check if a port is available (I borrowed this from `shiny::runApp()`, but I'd like to avoid a hard `httpuv` dependency if possible)